libsolidity/ast/ASTAnnotations.h

```diff
@@ -267,17 +267,22 @@ struct BinaryOperationAnnotation: ExpressionAnnotation
 
 enum class FunctionCallKind
 {
-	Unset,
 	FunctionCall,
 	TypeConversion,
 	StructConstructorCall
 };
 
 struct FunctionCallAnnotation: ExpressionAnnotation
 {
-	FunctionCallKind kind = FunctionCallKind::Unset;
```

Comment:
Is it possible to create a class similar to `std::optional`, but which causes an assertion failure if it is read before assignment and allows assignment exactly once?

Reply:
Sure. I'll write one.